godot-bindings/src/lib.rs

```diff
@@ -130,6 +130,17 @@ pub use prebuilt::*;
 // ----------------------------------------------------------------------------------------------------------------------------------------------
 // Common
 
+// List of minor versions with all relevant patch versions we may use in `cfg`s, this is used to emit appropriate `rustc-check-cfg` directives
+// to cargo.
+//
+// Note that when the patch version `0` then the patch number is not included in Godot versioning, i.e `4.1.0` is displayed as `4.1`.
+const KNOWN_API_VERSIONS: &[(u8, &[u8])] = &[
+    (0, &[0, 1, 2, 3, 4]),
+    (1, &[0, 1, 2, 3, 4]),
+    (2, &[0, 1, 2]),
+    (3, &[0]),
+];
```

Comment:
Probably easier to just store the highest known patch version for each minor one, i.e. a slice of `(u8, u8)` tuples.